src/server/main.py

```diff
@@ -56,15 +56,12 @@ def decorated(*args: tuple, **kwargs: dict) -> Response:
                 {"WWW-Authenticate": 'Basic realm ="Token is invalid!"'},
             )
 
-        try:
-            connection: Connection = getDatabaseConnection()
-            sql: str = "SELECT uuid, email FROM users WHERE uuid=?"
-            cursor: Cursor = connection.cursor()
-            cursor.execute(sql, (data["uuid"],))
-            user_information: list[tuple[str, str]] = cursor.fetchall()
-            connection.close()
-        except SqlError as e:
-            return make_response({"message": f"Database connection error: {e}"}, 500)
+        with DatabaseProvider.query(
+            "SELECT uuid, email FROM users WHERE uuid=?", (data["uuid"],)
+        ) as response:
+            if response.message is not Message.OK:
+                return make_response({"message": f"Internal error: {response.message}"}, 501)
+            user_information: list[tuple[str, str]] = response.data
```

Comment:
While mergining from master be aware that I've changed this query to only get uuid, not email, so you might need to change the type hinting.

Reply:
Sure, thanks for the heads-up